### PR TITLE
Add handle EncryptedWallet endpoint

### PR DESCRIPTION
**Pull request by Sajma27**
# Summary
A description of the changes proposed in the pull request.

Fixes #number

## Checklist
- [ ] Change is small and easy to review (please split big changes into multiple PRs)
- [ ] Change is consistent with architecture
- [ ] Change is consistent with test architecture
- [ ] Change is consistent with naming conventions
- [ ] New code is covered with tests
- [ ] Tests related to old code are updated
- [ ] Documentation is up to date with changes

### CODE REVIEW

File: universal-login-relayer/src/core/services/validators/EmailConfirmationValidator.ts [hunk 1]
```diff
@@ -7,9 +7,14 @@ export class EmailConfirmationValidator {
...
+  validateCode(emailConfirmation: EmailConfirmation, code: string) {
+    ensure(code === emailConfirmation.code, InvalidCode, code);
+  }
+
   validate(emailConfirmation: EmailConfirmation, email: string, code: string) {
     ensure(!emailConfirmation.isConfirmed, DuplicatedEmailConfirmation, email);
+    this.validateCode(emailConfirmation, code);
     ensure(moment(emailConfirmation.createdAt).add(this.codeDurationInMinutes, 'm').isAfter(), CodeExpired);
-    ensure(code === emailConfirmation.code, InvalidCode, code);
+    this.validateCode(emailConfirmation, code);
```
Comment (xitronix) [inline universal-login-relayer/src/core/services/validators/EmailConfirmationValidator.ts:18]: Duplicated check in `validate` function

File: universal-login-relayer/src/core/services/EncryptedWalletHandler.ts [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
   }
 
-  async handle(encryptedWallet: StoredEncryptedWallet) {
-    const emailConfirmation = await this.emailConfirmationsStore.get(encryptedWallet.email);
-    ensure(emailConfirmation.isConfirmed, EmailNotConfirmed, encryptedWallet.email);
-    return this.encryptedWalletsStore.add(encryptedWallet);
+  async handle(storedEncryptedWallet: StoredEncryptedWallet, code: string) {
+    const email = storedEncryptedWallet.email;
+    const emailConfirmation = await this.emailConfirmationStore.get(email);
+    ensure(emailConfirmation.isConfirmed, EmailNotConfirmed, email);
```
Comment (xitronix) [inline universal-login-relayer/src/core/services/EncryptedWalletHandler.ts:14]: Move to emailConfirmationValidator

File: universal-login-relayer/test/core/services/EncryptedWalletHandler.test.ts [hunk 3]
```diff
@@ -5,37 +5,46 @@ import {EncryptedWalletHandler} from '../../../src/core/services/EncryptedWallet
 import {TEST_ENCRYPTED_WALLET_JSON, StoredEncryptedWallet} from '@unilogin/commons';
 import {expect} from 'chai';
 import {createTestEmailConfirmation} from '../../testhelpers/createTestEmailConfirmation';
+import {EmailConfirmationValidator} from '../../../src/core/services/validators/EmailConfirmationValidator';
 
 describe('INT: EncryptedWalletHandler', () => {
-  const notConfirmedEmail = 'notConfirmed@email.com';
-  const confirmedEmail = 'confirmed@email.com';
+  const confirmedMail = 'confirmed@email.com';
```
Comment (xitronix) [inline universal-login-relayer/test/core/services/EncryptedWalletHandler.test.ts:11]: Why did you rename confirmedEmail  => confirmedMail?

File: universal-login-relayer/test/core/services/validators/EmailConfirmationValidator.unit.test.ts [hunk 4]
```diff
@@ -32,7 +32,6 @@ describe('UNIT: EmailConfirmationValidator', () => {
   it('Code expired', async () => {
     const expiredEmailConfirmation = {...emailConfirmation, createdAt: new Date('2020-07-09T13:00:00.000Z')};
     expect(() => validator.validate(expiredEmailConfirmation, email, code)).throws('Code expired');
-    (validator as any).codeDurationInMinutes = 60 * 24 * 5;
-    expect(validator.validate(expiredEmailConfirmation, email, code)).not.throws;
+    expect(validator.validate(emailConfirmation, email, code)).not.throws;
```
Comment (xitronix) [inline universal-login-relayer/test/core/services/validators/EmailConfirmationValidator.unit.test.ts:35]: It is not expected behavior of the test

File: universal-login-relayer/test/core/services/EncryptedWalletHandler.test.ts [hunk 5]
```diff
@@ -5,37 +5,46 @@ import {EncryptedWalletHandler} from '../../../src/core/services/EncryptedWallet
...
-    encryptedWalletHandler = new EncryptedWalletHandler(emailConfirmationsStore, encryptedWalletsStore);
+    encryptedWalletHandler = new EncryptedWalletHandler(emailConfirmationsStore, new EmailConfirmationValidator(), encryptedWalletsStore);
 
-    const emailConfirmation = createTestEmailConfirmation(notConfirmedEmail);
+    const emailConfirmation = createTestEmailConfirmation(notConfrimedEmail);
+    confirmationCode = emailConfirmation.code;
 
     storedEncryptedWallet = {
-      email: notConfirmedEmail,
+      email: notConfrimedEmail,
```
Comment (xitronix) [inline universal-login-relayer/test/core/services/EncryptedWalletHandler.test.ts:28]: should be `notConfirmedEmail`

File: universal-login-relayer/test/core/services/EncryptedWalletHandler.test.ts [hunk 6]
```diff
@@ -5,37 +5,46 @@ import {EncryptedWalletHandler} from '../../../src/core/services/EncryptedWallet
...
       ensName: emailConfirmation.ensName,
       walletJSON: TEST_ENCRYPTED_WALLET_JSON,
     };
 
     emailConfirmationsStore.add(emailConfirmation);
-    emailConfirmationsStore.add({...emailConfirmation, email: confirmedEmail, isConfirmed: true});
+    emailConfirmationsStore.add({...emailConfirmation, email: confirmedMail, isConfirmed: true});
+    emailConfirmationsStore.add({...emailConfirmation, email: invalidCodeEmail, isConfirmed: true});
```
Comment (xitronix) [inline universal-login-relayer/test/core/services/EncryptedWalletHandler.test.ts:35]: what is the difference between these twos? It can be just 
`emailConfirmationsStore.add({...emailConfirmation, email: confirmedMail, isConfirmed: true});`

File: universal-login-relayer/src/core/services/validators/EmailConfirmationValidator.ts (hunk 1 above)
Comment (Sajma27) [reply universal-login-relayer/src/core/services/validators/EmailConfirmationValidator.ts:18]: Done

File: universal-login-relayer/src/core/services/EncryptedWalletHandler.ts (hunk 2 above)
Comment (Sajma27) [reply universal-login-relayer/src/core/services/EncryptedWalletHandler.ts:14]: I think we should leave it like this

File: universal-login-relayer/test/core/services/EncryptedWalletHandler.test.ts (hunk 3 above)
Comment (Sajma27) [reply universal-login-relayer/test/core/services/EncryptedWalletHandler.test.ts:11]: Done

File: universal-login-relayer/test/core/services/validators/EmailConfirmationValidator.unit.test.ts (hunk 4 above)
Comment (Sajma27) [reply universal-login-relayer/test/core/services/validators/EmailConfirmationValidator.unit.test.ts]: Done

File: universal-login-relayer/test/core/services/EncryptedWalletHandler.test.ts (hunk 5 above)
Comment (Sajma27) [reply universal-login-relayer/test/core/services/EncryptedWalletHandler.test.ts:28]: Done

File: universal-login-relayer/test/core/services/EncryptedWalletHandler.test.ts (hunk 6 above)
Comment (Sajma27) [reply universal-login-relayer/test/core/services/EncryptedWalletHandler.test.ts:35]: Done

File: universal-login-relayer/src/core/services/EncryptedWalletHandler.ts (hunk 2 above)
Comment (Sajma27) [reply universal-login-relayer/src/core/services/EncryptedWalletHandler.ts]: Done

File: universal-login-relayer/src/core/services/validators/EmailConfirmationValidator.ts [hunk 7]
```diff
@@ -1,15 +1,23 @@
...
 import {ensure, EmailConfirmation} from '@unilogin/commons';
-import {DuplicatedEmailConfirmation, InvalidCode, CodeExpired} from '../../utils/errors';
+import {DuplicatedEmailConfirmation, InvalidCode, CodeExpired, EmailNotConfirmed} from '../../utils/errors';
 
 export class EmailConfirmationValidator {
   constructor(
     private codeDurationInMinutes: number = 5,
   ) {}
 
+  validateEmail(emailConfirmation: EmailConfirmation) {
```
Comment (xitronix) [inline universal-login-relayer/src/core/services/validators/EmailConfirmationValidator.ts:10]: `isConfirmed` or `validateIsConfirmed`. We don't validate email here but if it is confirmed

Comment (Sajma27) [reply universal-login-relayer/src/core/services/validators/EmailConfirmationValidator.ts]: Done

File: universal-login-relayer/src/http/routes/wallet.ts [hunk 8]
```diff
@@ -112,5 +120,13 @@ export default (
     futureWalletHandling(futureWalletHandler),
   ));
 
+  router.post('/encrypted', asyncHandler(
```
Comment (xitronix) [inline universal-login-relayer/src/http/routes/wallet.ts:123]: Add test for this endpoint

Comment (Sajma27) [reply universal-login-relayer/src/http/routes/wallet.ts:123]: Done
